lib/config/configuration_test.go

```diff
@@ -1363,3 +1363,37 @@ func TestDatabaseFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckAndSetDefaults(t *testing.T) {
+	tests := []struct {
+		comment      string
+		formatConfig []string
+		hasError     bool
+	}{
+		{
+			comment:      "invalid key (does not exist)",
+			formatConfig: []string{"level", "invalid key"},
+			hasError:     true,
+		},
+		{
+			comment:      "valid keys and formatting",
+			formatConfig: []string{"level", "component", "message"},
+			hasError:     false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.comment, func(t *testing.T) {
+			formatter := &textFormatter{
+				InitFormat: tt.formatConfig,
+			}
+			err := formatter.CheckAndSetDefaults()
+			if tt.hasError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+
+}
```

Comment:
The way I was implementing it was so that order could be taken into account as well. I can implement in such a way that ordering does not matter (that does make more sense since the config file takes a list). I can implement it such a way that timestamp is always first and caller is last. Does that sound good going forward with implementation?

Reply:
Yes, it feels more expected behavior to me - e.g. if specified, put timestamp first, caller last and maybe remove message as a user-specified field since log entry needs to output its message anyways.